include/vcpkg/base/messages.h

```diff
@@ -63,7 +63,7 @@ namespace vcpkg
             return *this;
         }
         template<class Message, class... Args>
-        LocalizedString& append(Message m, const Args&... args)
+        constexpr LocalizedString& append(Message m, const Args&... args)
```

Comment:
This requires C++20 (and even then a very recent C++20 compiler since constexpr string is taking a long time to land), so I don't think we can do that. https://en.cppreference.com/w/cpp/compiler_support claims gcc 12 / msvc 2022 / clang 15. (Unavailable for Apple Clang entirely)

Reply:
Removed constexpr here.

Reply:
@BillyONeal Now I see where confusion about C++ 20 came from: I already removed constexpr here. Note that this change had no impact on my approach.